src/OSPSuite.Core/Commands/ChangeObservedDataMetaDataCommand.cs

```diff
@@ -21,11 +22,21 @@ protected override void ExecuteWith(IOSPSuiteExecutionContext context)
       {
          changeMetaDataInRepository();
 
+         updateMolecule(context);
+
          Description = Command.SetMetaDataChangedCommandDescription(_metaDataChanged.OldName, _metaDataChanged.OldValue, _metaDataChanged.NewName, _metaDataChanged.NewValue);
          SetBuildingBlockParameters(context);
          context.PublishEvent(new ObservedDataMetaDataChangedEvent(_observedData));
       }
 
+      private void updateMolecule(IOSPSuiteExecutionContext context)
+      {
+         if(_metaDataChanged.NewName != Constants.ObservedData.MOLECULE) return;
+
+         var service = context.Resolve<IObservedDataTask>();
```

Comment:
yes. I would argue that we should call service observedDataTask however:) not a big deal but like variable to be explicit

Reply:
corrected